sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_client.py

```diff
@@ -6,73 +6,75 @@
 # Changes may cause incorrect behavior and will be lost if the code is regenerated.
 # --------------------------------------------------------------------------
 
-from typing import TYPE_CHECKING
+from copy import deepcopy
+from typing import Any, TYPE_CHECKING
 
-from azure.core import PipelineClient
 from msrest import Deserializer, Serializer
 
+from azure.core import PipelineClient
+from azure.core.rest import HttpRequest, HttpResponse
+
+from ._configuration import MetricsAdvisorClientConfiguration
+from ._operations import MetricsAdvisorClientOperationsMixin
+from .models import _models as models
+
 if TYPE_CHECKING:
     # pylint: disable=unused-import,ungrouped-imports
-    from typing import Any
-
     from azure.core.credentials import TokenCredential
-    from azure.core.pipeline.transport import HttpRequest, HttpResponse
-
-from ._configuration import MetricsAdvisorConfiguration
-from .operations import MetricsAdvisorOperationsMixin
-from . import models
 
 
-class MetricsAdvisor(MetricsAdvisorOperationsMixin):
+class MetricsAdvisorClient(MetricsAdvisorClientOperationsMixin):
     """Microsoft Azure Metrics Advisor REST API (OpenAPI v2).
 
-    :param credential: Credential needed for the client to connect to Azure.
-    :type credential: ~azure.core.credentials.TokenCredential
-    :param endpoint: Supported Cognitive Services endpoints (protocol and hostname, for example: https://:code:`<resource-name>`.cognitiveservices.azure.com).
+    :param endpoint: Supported Cognitive Services endpoints (protocol and hostname, for example:
+     https://:code:`<resource-name>`.cognitiveservices.azure.com). Required.
     :type endpoint: str
+    :param credential: Credential needed for the client to connect to Azure. Required.
+    :type credential: ~azure.core.credentials.TokenCredential
     """
 
-    def __init__(
-        self,
-        credential,  # type: "TokenCredential"
-        endpoint,  # type: str
-        **kwargs  # type: Any
-    ):
-        # type: (...) -> None
-        base_url = '{endpoint}/metricsadvisor/v1.0'
-        self._config = MetricsAdvisorConfiguration(credential, endpoint, **kwargs)
-        self._client = PipelineClient(base_url=base_url, config=self._config, **kwargs)
+    def __init__(self, endpoint: str, credential: "TokenCredential", **kwargs: Any) -> None:
```

Comment:
This is an intentional breaking change?

Reply:
That's not exposed to users, the clients in _patch.py are the ones exposed